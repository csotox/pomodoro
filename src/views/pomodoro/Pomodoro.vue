<template>
  <section class="pomodoro">
    <div class="application__content pomodoro__content">
      <h2 class="pomodoro__title">Pomodoro</h2>
      <h1 class="pomodoro__type">{{ pomodoroType }}</h1>

      <Timer class="pomodoro__timer" :time="pomodoroTime" />

      <RoundedButton @click="onClick()">
        <ion-icon name="close"></ion-icon>
      </RoundedButton>
    </div>
  </section>
</template>

<script>
import Timer from '@/components/Timer.vue';
import PomodoroTypes from './pomodoroTypes';
import RoundedButton from '@/components/RoundedButton.vue';

export default {
  name: 'Pomodoro',
  components: {
    Timer,
    RoundedButton,
  },
  computed: {
    pomodoroType() {
      return this.$route.params.type;
    },
    pomodoroTime() {
      return PomodoroTypes[this.pomodoroType].time;
    },
  },
  methods: {
    onClick() {
      this.$router.push({ name: 'pomodoro', params: { type: 'rest' } });
    },
  },
};
</script>

<style lang="scss" scoped>
.pomodoro {
  background-color: var(--application-color-primary);
  color: var(--application-color-text);
}

.pomodoro__content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.pomodoro__title {
  font-size: 1rem;
  text-transform: uppercase;
}

.pomodoro__type {
  margin-bottom: 2rem;
  font-size: 4rem;
  font-weight: 500;
  letter-spacing: 1px;
  text-transform: uppercase;
}

.pomodoro__timer {
  margin-bottom: 3rem;
  font-size: 8rem;
  font-weight: 800;
}

ion-icon {
  font-size: 64px;
}
</style>
