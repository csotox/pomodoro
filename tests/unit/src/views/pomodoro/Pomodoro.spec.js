import { mount } from '@vue/test-utils';
import Pomodoro from '@/views/pomodoro/Pomodoro.vue';
import RoundedButton from '@/components/RoundedButton.vue';
import PomodoroTypes from '@/views/pomodoro/pomodoroTypes';

function createPomodoroView(options = {}) {
  return mount(Pomodoro, {
    stubs: ['ion-icon'],
    mocks: {
      $route: {
        params: {
          type: 'work',
        },
      },
      ...options.mocks,
    },
  });
}

describe('Pomodoro.vue', () => {
  it('should render properly', () => {
    const pomodoro = createPomodoroView();
    expect(pomodoro).toMatchSnapshot();
  });

  describe('Computed', () => {
    describe('pomodoroType', () => {
      it('should return type parameter from route', () => {
        const pomodoro = createPomodoroView();
        expect(pomodoro.vm.pomodoroType).toEqual('work');
      });
    });

    describe('pomodoroTime', () => {
      it('should return type parameter from route', () => {
        const pomodoro = createPomodoroView();
        expect(pomodoro.vm.pomodoroTime).toEqual(PomodoroTypes.work.time);
      });
    });
  });

  describe('Methods', () => {
    describe('onClick', () => {
      it('should redirect to rest pomodoro when close button is clicked', () => {
        const onClickSpy = jest.fn();
        const pomodoro = createPomodoroView({
          mocks: {
            $router: {
              push: onClickSpy,
            },
          },
        });

        pomodoro.find(RoundedButton).trigger('click');

        expect(onClickSpy).toHaveBeenCalledWith(
          { name: 'pomodoro', params: { type: 'rest' } },
        );
      });
    });
  });
});
