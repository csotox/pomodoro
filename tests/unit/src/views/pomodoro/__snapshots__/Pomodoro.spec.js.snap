// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pomodoro.vue should render properly 1`] = `
<section class="pomodoro">
  <div class="application__content pomodoro__content">
    <h2 class="pomodoro__title">Pomodoro</h2>
    <h1 class="pomodoro__type">work</h1>
    <section class="pomodoro__timer">
      00:00
    </section> <button class="rounded-button">
      <ion-icon-stub name="close"></ion-icon-stub>
    </button>
  </div>
</section>
`;
